src/test/fixtures/fake-api-token-store.ts

```diff
@@ -39,11 +38,8 @@ export default class FakeApiTokenStore
     }
 
     async get(key: string): Promise<IApiToken> {
-        const token = this.tokens.find((t) => t.secret === key);
-        if (token) {
-            return token;
-        }
-        throw new NotFoundError(`Could not find token with secret ${key}`);
+        // @ts-expect-error get can return undefined. See api-token-store.e2e.test.ts
+        return this.tokens.find((t) => t.secret === key);
```

Comment:
@nunogois just FYI while looking into this, I also found that this fake store was wrongly defined. I've tested how the real one behaves in src/test/e2e/stores/api-token-store.e2e.test.ts